### PR TITLE
feat: add some warnings in IIFE and CJS.

### PR DESCRIPTION
**Pull request by 7086cmd**
resolves #1855.

1. 
```
No name was provided for external module "node:path" in "output.globals" – guessing "node_path".
```
2.
```
If you do not supply "output.name", you may not be able to access the exports of an IIFE bundle.
```
3. 
```
Entry module "main.js" is using named and default exports together. Consumers of your bundle will have to use `chunk.default` to access the default export, which may not be what you want. Use `output.exports: "named"` to disable this warning.
```

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
Can you see esbuild or rollup behavior?

**Comment by 7086cmd** [comment]
> Can you see esbuild or rollup behavior?

[Here](https://rollupjs.org/repl/?version=4.20.0&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMGpvaW4lMjAlN0QlMjBmcm9tJTIwJ25vZGUlM0FwYXRoJyU1Q25pbXBvcnQlMjB1cmwlMjBmcm9tJTIwJ3VybCclNUNuJTVDbmltcG9ydCgnLiUyRmZpbGUnKS50aGVuKChpbXBvcnRlZCklMjAlM0QlM0UlMjAlN0IlNUNuJTIwJTIwYXNzZXJ0LnN0cmljdEVxdWFsKGltcG9ydGVkLnNoYXJlZCUyQyUyMCdzaGFyZWQnKSU1Q24lN0QpJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMG1haW4lMjAlM0QlMjAnbWFpbiclNUNuJTVDbmV4cG9ydCUyMGNvbnN0JTIwcGF0aCUyMCUzRCUyMCU3QiUyMGpvaW4lMjAlN0QlNUNuZXhwb3J0JTIwJTdCJTIwdXJsJTIwJTdEJTVDbiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwMiUzQiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBzaGFyZWQlMjAlM0QlMjAnc2hhcmVkJyU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyZmlsZS5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJlc01vZHVsZSUyMiUzQSUyMmlmLWRlZmF1bHQtcHJvcCUyMiUyQyUyMmV4cG9ydHMlMjIlM0ElMjJhdXRvJTIyJTJDJTIyZm9ybWF0JTIyJTNBJTIyaWlmZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlMjJub2RlJTNBcGF0aCUyMiUzQSUyMiUyMiU3RCUyQyUyMmlubGluZUR5bmFtaWNJbXBvcnRzJTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjIlMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0F0cnVlJTdEJTdE) is a gather version in Rollup.

**Comment by IWANABETHATGUY** [comment]
I assume they are both warning, but it display in `Error`

**Comment by 7086cmd** [comment]
I'll fix them soon. It is a bug.

File: crates/rolldown/src/ecmascript/format/iife.rs [hunk 1]
```diff
@@ -37,25 +38,30 @@ pub fn render_iife(
...
   };
-  let export_mode = determine_export_mode(&ctx.options.exports, entry_module, &export_items)?;
+  let export_mode = determine_export_mode(ctx, entry_module, &export_items)?;
   let named_exports = matches!(&export_mode, OutputExports::Named);
 
   let (import_code, externals) = render_iife_chunk_imports(ctx);
 
   let (input_args, output_args) =
-    render_iife_arguments(&externals, &ctx.options.globals, has_exports && named_exports);
+    render_iife_arguments(ctx, &externals, &ctx.options.globals, has_exports && named_exports);
```
Comment (IWANABETHATGUY) [inline crates/rolldown/src/ecmascript/format/iife.rs]: Since you already pass the `ctx`, you don't need to pass `ctx.options.globals`.

Comment (7086cmd) [reply crates/rolldown/src/ecmascript/format/iife.rs]: Resolved.

File: crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json]: Is this test case related?

Comment (7086cmd) [reply crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json]: Nope, it seems that I brought it from the previous branch before this :) Is it necessary to remove it?

Comment (IWANABETHATGUY) [reply crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json]: Let's remove it first.

**Comment by IWANABETHATGUY** [comment]
LGTM

File: crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json (hunk 2 above)
Comment (7086cmd) [reply crates/rolldown/tests/rolldown/misc/footer/cjs/_config.json]: Okay, I'll remove it later (hanging out now).

**Review by underfin — APPROVED** [review]

**Review by IWANABETHATGUY — APPROVED** [review]
